From 5f233774a854835fbac4c080e33e2b1aedb35644 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 27 Jul 2021 11:35:16 +0300
Subject: [PATCH] [SW-49810] ras: ignore reset reason after reboot

Reset reason is transferred from preboot to Linux by SPSRAM so that the
BMC is able to read the watchdog error if exists which caused the reset.
The problem is the same reset cause stored in SPSRAM is transferred back
to preboot which sends it back to Linux, indicating the same reset
reason again and again.

Fixed by overriding reset reason on SPSRAM when preboot loads. Any reset
cause indication needs to come from either the driver or HW.

Change-Id: I6f76a040d21067f6346970edd8f75162254dc454
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 5b1f523..6ebcdae 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -359,6 +359,12 @@
 		else
 			preboot_puts_hex("Ignoring fw info with version: ",
 					 fw_info_local.version, false);
+
+		/*
+		 * Ignore reset cause that came from Linux as it is invalid to
+		 * current reset sequence.
+		 */
+		fw_info.reset_cause = HL_FW_RESET_CAUSE_UNKNOWN;
 	}
 }
 
